### PR TITLE
hash funtions

### LINKED ISSUE #98: Feature: 17.4.6 Hash functions
https://www.w3.org/TR/sparql11-query/#func-hash

- [x] 17.4.6.1 MD5
- [x] 17.4.6.2 SHA1
- [x] 17.4.6.3 SHA256
- [x] 17.4.6.4 SHA384
- [x] 17.4.6.5 SHA512

### PR DESCRIPTION
**Pull request by mcb5637**
implementing #98 via OpenSSL

### CODE REVIEW

**Comment by bigerl** [comment]
I've cleaned the GitHub Actions cache. The separation between different configurations was beached somehow. Should be fine again, now. 
I also revered the changes to the CMake files regarding OpenSSL. 

One build problem with gcov remains. Please investigate that locally in your local environment. Right now, gcov is run but not uploaded. So, if there is no simple solution, just deactivate gcov in the CI.

File: src/rdf4cpp/rdf/Literal.cpp [hunk 1]
```diff
@@ -1556,6 +1558,48 @@ Literal Literal::substr(Literal const &start, Literal const &len, Node::NodeStor
     return this->substr(start_ix, len_ix, node_storage);
 }
 
+Literal Literal::hash_with(const char *alg, NodeStorage &node_storage) const {
+    if (!this->is_string_like())
```
Comment (liss-h) [inline src/rdf4cpp/rdf/Literal.cpp]: This passes for `rdf:langString` and `xsd:string` but as far as I can tell these functions are only supposed to work for `xsd:string`

File: src/rdf4cpp/rdf/Literal.hpp [hunk 2]
```diff
@@ -920,6 +920,43 @@ class Literal : public Node {
                                  Literal const &len = Literal::make_typed_from_value<datatypes::xsd::Double>(std::numeric_limits<datatypes::xsd::Double::cpp_type>::infinity()),
                                  NodeStorage &node_storage = NodeStorage::default_instance()) const noexcept;
 
+private:
+    /**
+     * hash via OpenSSL
+     * @param alg OpenSSL hash algorithm identifier
```
Comment (liss-h) [inline src/rdf4cpp/rdf/Literal.hpp]: A link to the options for `alg` would be nice here

**Review by liss-h — CHANGES_REQUESTED** [review]
Thank you; again only small details to fix

**Review by liss-h — APPROVED** [review]

**Pull request merged**